### ISSUE TITLE
Add Fargate install docs for Sourcegraph

### ISSUE THREAD

**Issue by KattMingMing** [sourcegraph/sourcegraph#2568]
A user requested documentation for how to install Sourcegraph via Fargate.

**Comment by ryan-blunden** [comment]
It's definitely doable and is something that could be prioritized for 3.3.

This would only be available to enterprise customers because though because Fargate are designed to be stateless, therefore requiring an external PostgreSQL instance.

**Comment by joshma** [comment]
(I'm the user who mentioned Fargate support would be nice.) To confirm, an external DB instance is only available on enterprise, even if we're self hosting? I was hoping to hook it up to an RDS instance. Is the Redis instance just for caching (is it okay for it to be destroyed from time to time)? Can we host it via a second container?

**Comment by sqs** [comment]
Yes, an external DB is an a enterprise feature because we’ve found that it is a strong indicator that a company is using it in production at large scale with large company-esque needs, which is where we think it’s fair to charge. If this doesn’t describe your situation, just drop us an email (or explain here if you want) and we’ll see what to do.

**Comment by joshma** [comment]
Hm, for us it's not about scale, but operational costs - a multi-az t3.small RDS instance with 35-day backup is about $50-$60/mo, and that means I don't need to operate a special instance that folks on the team need special knowledge about (SSH-in, docker pull, be sure to not terminate this because the DB goes away with it, let's start taking EBS snapshots as a backup...).

Running this on something like Fargate is pretty much impossible without an external instance I believe - unless Fargate were to support re-mounting EBS volumes across tasks, it's only inevitable for a Fargate container to get replaced, and along w/ it the database.

**Comment by sqs** [comment]
@joshma 	Thanks for the context. Makes sense. @KattMingMing will follow up over email.

**Comment by ahudelson** [comment]
Has any progress been made on this?  I found Sourcegraph yesterday, and it looks like it could be a good solution for us.  I'd really like to prop this up in Fargate with an RDS database for similar reasons to what @joshma statec on March 5, 2019.  It would be the most cost-effective, hands off solution for us.

**Comment by rfulwell** [comment]
Likewise, we're interested in this for a low-effort internal hosting solution until Sourcegraph provides a SaaS offering for private repos. Otherwise we're looking at spinning up another EKS cluster just for this which is a non-trivial amount of work for our small team to properly render all the bits as Terraform, etc.
